### PR TITLE
Implement empty?, member?, size and filter methods

### PR DESCRIPTION
**Pull request by brandonhamilton**
Implement empty?, member? and size methods.

### CODE REVIEW

File: lib/blocking_queue.ex [hunk 1]
```diff
@@ -144,6 +144,11 @@ defmodule BlockingQueue do
     {:reply, :queue.member(item, queue), {max, queue}}
   end
 
+  # remove all items using predicate function
+  def handle_call({:filter, f}, _, {max, queue}) do
+    {:reply, nil, {max, :queue.filter(f, queue)}}
+  end
```
Comment (joekain) [inline lib/blocking_queue.ex]: What if the the queue is full and there are waiting pushers?  If the `:queue.filter` call returns a queue that isn't full then don't we have to accept any waiting pushes (and potentially filter them)?

Comment (brandonhamilton) [inline lib/blocking_queue.ex]: You are absolutely correct, I have failed to account for any push waiters as my use case was with infinite queues. I have pushed a new commit that takes this into account

**Comment by brandonhamilton** [comment]
I do not know why SemaphoreCI is failing the tests, as I can successfully run them locally (Elixir 1.3.3)

**Comment by joekain** [comment]
Thanks for the update @brandonhamilton.

I just looked into the CI failures and it looks like Semaphore deprecated the versions of Elixir I was using.  That's my fault, I haven't been keeping this repo up to date.  I just updated the CI settings and picked a supported version of Elixir for the build and tests.  The tests now pass.  The docs step fails but for a reason unrelated to this change.  I'll fix that later today.

Since this change passes I'm going to go ahead and merge it.

**Pull request merged**